src/parser/jobs/ast/modules/ArcanaTracking/CritAndDHRates.tsx

```diff
@@ -0,0 +1,422 @@
+			filter<Event>()
+				.source(oneOf(this.pets))
+				.type('damage'),
+			this.petDamage
+		)
+
+		//statuses
+		this.addEventHook(
+			filter<Event>()
+				.target(oneOf(this.playerControlled))
+				.type('statusApply')
+				.status(oneOf(this.critPartyStatuses)),
+			this.critPartyApply
+		)
+		this.addEventHook(
+			filter<Event>()
+				.target(oneOf(this.playerControlled))
+				.type('statusRemove')
+				.status(oneOf(this.critPartyStatuses)),
+			this.critPartyRemove
+		)
+
+		this.addEventHook(
+			filter<Event>()
+				.target(oneOf(this.playerControlled))
+				.type('statusApply')
+				.status(oneOf(this.DHPartyStatuses)),
+			this.DHPartyApply
+		)
+		this.addEventHook(
+			filter<Event>()
+				.target(oneOf(this.playerControlled))
+				.type('statusRemove')
+				.status(oneOf(this.DHPartyStatuses)),
+			this.DHPartyRemove
+		)
+
+		//note no target since assumed only enemies receive these statuses
+		this.addEventHook(
+			filter<Event>()
+				.type('statusApply')
+				.status(oneOf(this.critEnemyStatuses)),
+			this.critEnemyApply
+		)
+		this.addEventHook(
+			filter<Event>()
+				.type('statusRemove')
+				.status(oneOf(this.critEnemyStatuses)),
+			this.critEnemyRemove
+		)
+
+		//complete
+		this.addEventHook('complete', this.onComplete)
+	}
+
+	private damage(event: Events['damage']) {
+		//statuses are given
+		if (event.cause.type === 'status'
+				&& (event.directHitRate !== 0 && event.directHitRate != null
+				|| event.expectedCritRate !== 0 && event.expectedCritRate != null)) {
+			this.statusEvent.push({
+				actor: event.source,
+				timestamp: event.timestamp,
+				action: event.cause.status,
+				critRate: event.expectedCritRate ?? 0 * ROUNDING_FACTOR, //crit rate is stated in 0.1
+				DHRate: event.directHitRate ?? 0,
+			})
+			return
+		}
+
+		if (event.targets.length === 0) { return }
+		const action = event.cause.type === 'action' ? event.cause.action : event.cause.status
+		const abilityCritModifier = this.autoCritAbilities.includes(action) ? 1 : 0
+		const abilityDHModifier = this.autoDHAbilities.includes(action) ? 1 : 0
+		event.targets.forEach(target => {
+			//verify it is initialized
+			if (this.critEnemyModifierCurrent[target.target] === undefined) {
+				this.critEnemyModifierCurrent[target.target] = {
+					currentModifier: 0,
+				}
+			}
+			this.damageEvent.push(
+				{
+					actor: event.source,
+					timestamp: event.timestamp,
+					action: action,
+					critMod: this.critPartyModifierCurrent[event.source].currentModifier + this.critEnemyModifierCurrent[target.target].currentModifier + abilityCritModifier,
+					DHMod: this.DHPartyModifierCurrent[event.source].currentModifier + abilityDHModifier,
+					sourceModifier: target.sourceModifier,
+				}
+			)
+		})
+	}
+
+	private petDamage(event: Events['damage']) {
+		const owner: Actor['id'] | undefined = this.actors.get(event.source)?.owner?.id
+		//don't consider statuses as already considered in Damage and not expecting pets to have ticks
```

Comment:
I think this is an unsafe assumption since pets have had DoTs in the past.

Reply:
Fair
Updated to include pet DoTs
See below commit